modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/exceptions/XMLFileOperationException.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.common.exceptions;
+
+/**
+ * Exception class for property/provisioning XML file reading failures.
+ */
+public class XMLFileOperationException extends Exception {
```

Comment:
To me, XMLFileOperationException doesn't make much sense. If there was some issue in XML parsing, etc, some appropriate exception such as ParserException, would be thrown. We can perhaps, catch them and throw some other high-level exception like, WAPProvisioningException which would be much elegant, IMO.